### PR TITLE
[MM-15302] Migrate `Preference.Delete` to Sync by default #10715 

### LINKED ISSUE #10715: [MM-15302] Migrate "Preference.Delete" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15302)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `Delete` in the `Preference` store.

The expected way to implement it is, go to the `Preference` store implementation in the `store/sqlstore/` directory, modify the method `Delete` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by Inconnu08**
Summary
Migrate `Preference` to Sync by default removing the `return.Do` wrapper.

Ticket Link
Fixes #10715

### CODE REVIEW

**Comment by Inconnu08** [comment]
Accidentally closed #11057

File: store/sqlstore/preference_store.go [hunk 1]
```diff
@@ -236,18 +236,21 @@ func (s SqlPreferenceStore) IsFeatureEnabled(feature, userId string) (bool, *mod
...
-			WHERE
-				UserId = :UserId
-				AND Category = :Category
-				AND Name = :Name`, map[string]interface{}{"UserId": userId, "Category": category, "Name": name}); err != nil {
-			result.Err = model.NewAppError("SqlPreferenceStore.Delete", "store.sql_preference.delete.app_error", nil, err.Error(), http.StatusInternalServerError)
-		}
-	})
+func (s SqlPreferenceStore) Delete(userId, category, name string) *model.AppError {
+	query :=
+		`DELETE FROM Preferences
```
Comment (mounicapaladugu) [inline store/sqlstore/preference_store.go]: probably you can remove additional tab here??

**Comment by jespino** [comment]
@Inconnu08 you didn't regenerate the mocks for the store.

File: store/sqlstore/preference_store.go (hunk 1 above)
Comment (Inconnu08) [reply store/sqlstore/preference_store.go]: @mounicapaladugu But it's better readability?

**Comment by jespino** [comment]
@Inconnu08 Thanks! :sparkles:

**Pull request merged**